components/raftstore/src/store/snap.rs

```diff
@@ -2094,18 +2094,29 @@ impl TabletSnapManager {
         PathBuf::from(&self.base).join(prefix)
     }
 
-    pub fn delete_snapshot(&self, key: &TabletSnapKey) -> bool {
-        let path = self.tablet_gen_path(key);
-        if path.exists() && let Err(e) = file_system::trash_dir_all(&path) {
-            error!(
-                "delete snapshot failed";
-                "path" => %path.display(),
-                "err" => ?e,
-            );
-            false
-        } else {
-            true
+    pub fn delete_snapshot(&self, region_id: u64, to_peer: u64) -> Result<()> {
+        for f in file_system::read_dir(&self.base)? {
+            let entry = f?;
+            let ft = entry.file_type()?;
+            if ft.is_file() {
+                continue;
+            }
+            let os_name = entry.file_name();
+            let name = os_name.to_str().unwrap().to_string();
+
+            let parts = name
+                .split('_')
+                .skip(1)
+                .map(|s| s.parse().unwrap())
+                .collect::<Vec<u64>>();
+            if parts.len() < 4 {
```

Comment:
In which cases it will be less than 4?

Reply:
none, but user may create dir in this directory.